### PR TITLE
Send task execution metadata to out-core plugin

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Send task execution metadata to out-core plugin when creating a task.

This depends on https://github.com/flyteorg/flyteidl/pull/422

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [x] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
As described in https://github.com/flyteorg/flyte/issues/3824, for certain type of use cases, the task execution metadata would be desired to have.

## Tracking Issue
Closes https://github.com/flyteorg/flyte/issues/3824

## Follow-up issue
_NA_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -135,3 +135,6 @@ require (
 )
 
 replace github.com/aws/amazon-sagemaker-operator-for-k8s => github.com/aws/amazon-sagemaker-operator-for-k8s v1.0.1-0.20210303003444-0fb33b1fd49d
+
+// TODO: remove this when flyteidl is ready
```
Comment (honnix) [inline go.mod:139]: @honnix Remove this after https://github.com/flyteorg/flyteidl/pull/422 is merged.

**Comment by hamersaw** [comment]
@honnix sorry about the delay, can you update this PR with the flyteidl merge?

**Comment by honnix** [comment]
@hamersaw I fixed that.

**Review by hamersaw — APPROVED** [review]
